Engine/Spells/CastSpellInfo.cpp

```diff
@@ -851,13 +773,14 @@ void CastSpellInfoHelpers::_427E01_cast_spell() {
                     default:
                         assert(false);
                 }
-                if (!pPlayer->CanCastSpell(uRequiredMana)) break;
-                v730c = &pParty->pPlayers[pCastSpell->uPlayerID_2]
-                             .pInventoryItemList[spell_targeted_at - 1];
+                if (!pPlayer->CanCastSpell(uRequiredMana))
+                    break;
+
+                v730c = &pParty->pPlayers[pCastSpell->uPlayerID_2].pInventoryItemList[pCastSpell->spell_target_pid];
                 _item = &pItemsTable->pItems[v730c->uItemID];
                 v730c->UpdateTempBonus(pParty->GetPlayingTime());
-                if (v730c->uItemID < 64 ||
-                    v730c->uItemID > 65 && !v730c->IsBroken() &&
+                if (v730c->uItemID < ITEM_BLASTER || v730c->uItemID > ITEM_LASER_RIFLE &&
```

Comment:
Looks like we have missing parentheses here.

(v730c->uItemID < ITEM_BLASTER || v730c->uItemID > ITEM_LASER_RIFLE) && !v730c->IsBroken() etc...